packages/move/sources/vesting.move

```diff
@@ -25,14 +25,29 @@ module deployment_addr::vesting {
     /// User try to claim zero
     const ERR_AMOUNT_ZERO: u64 = 10;
 
-    struct VestingStream has key, store, drop {
+    // FIX: key ability is not needed for VestingStream since it isn't keyed to an address in global storage
+
+    // FIX: current implementation treats cliff as a delay period before linear vesting begins. Which is kind of redundant since the start_time can be specified
+    // For a proper vesting cliff implementation, VestingStream should include a cliff_amount field
+    // that represents tokens immediately unlocked when cliff is reached. Current implementation
+    // seems to always start linear vesting from 0 at cliff time, whereas typically some percentage
+    // of tokens should unlock immediately at cliff time(in your case that percentage is effectively 0).
+
+    struct VestingStream has store, drop {
         amount: u64,
         claimed_amount: u64,
         start_time: u64,
         cliff: u64,
         duration: u64
     }
 
+    // FIX: it seems like your implementation requires a new vesting module to be deployed for every FA to be streamed
+    // and it also only supports a single VestingStream for a given address.
+    // To generalize this I would instead allow anyone to create arbitrary VestingStream object instances i.e. Object<VestingStream>
+    // The owner of the Object<VestingStream> is essentially the admin and can do admin actions
+    // #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    // And the object signer associated with the Object<VestingStream> can be used to withdraw funds from the fungible store for that stream
```

Comment:
Yes based on my assumptions I created the module to be redeployed for each FA, and allowing only one active stream per address.
With your suggested approach, I am curious how a user/beneficiary can easily locate his streams, since the Object<VestingStream> will be owned by the admin and not the beneficiary.

Reply:
A similar answer to this https://github.com/mshakeg/canopy-vesting-contract/pull/1#discussion_r1900231138

So an indexing service would make all of this data available, and also do aggregations, including maintain a list of all streams that a given address is a beneficiary off, and the UI would query this service to get that list. You'd just have to ensure that the module emits all the required events in order for this indexer to have all data available.

What do you think?